classes/backend/api.php

```diff
@@ -7,27 +7,45 @@
 class api {
     private string $api_url;
 
+    public const TASK_STATE_NEW = 'new';
+    public const TASK_STATE_WAITING_FOR_RESULTS = 'waiting_for_results';
+    public const TASK_STATE_RETRIEVE_SUBMISSION = 'retrieve_submission';
+    public const TASK_STATE_DONE = 'done';
+
     public function __construct(string $api_url) {
         $this->api_url = $api_url;
     }
 
+    private function clean_url(string $url) {
+        $offset = strlen('http://');
+
+        if (strncmp('https', $url, strlen('https')) === 0)
+            $offset++;
+
+        $protocol_free = substr($url, $offset);
+        $clean_url_part = preg_replace('/(\/+)/', '/', $protocol_free);
+
+        return substr($url, 0, $offset) . $clean_url_part;
```

Comment:
This could probably be turned into `preg_replace('#https?://(*SKIP)(*FAIL)|/+#', '/', $url)`, but it needs to be properly tested.